### PR TITLE
updated p2p api send and receive requests for flexible order time sel…

### PR DESCRIPTION
**Pull request by weng-deriv**
Cu link: https://app.clickup.com/t/20696747/P2PS-638

### CODE REVIEW

**Review by Reza-deriv — APPROVED** [review]

File: lib/api/response/p2p_advert_info_response_result.dart [hunk 1]
```diff
@@ -1197,16 +1149,11 @@ abstract class FieldsPropertyModel {
...
-    required TypeEnum type,
-    required String value,
-  }) : super(
-          displayName: displayName,
-          required: required,
-          type: type,
-          value: value,
-        );
+    required super.displayName,
+    required super.required,
```
Comment (abedelaziz-deriv) [inline lib/api/response/p2p_advert_info_response_result.dart:1153]: `required` is a reserved word, shouldn't it be renamed to something else ? @weng-deriv

File: lib/api/response/p2p_advert_update_response_result.dart [hunk 2]
```diff
@@ -1036,16 +993,11 @@ abstract class FieldsPropertyModel {
...
-    required TypeEnum type,
-    required String value,
-  }) : super(
-          displayName: displayName,
-          required: required,
-          type: type,
-          value: value,
-        );
+    required super.displayName,
+    required super.required,
```
Comment (abedelaziz-deriv) [inline lib/api/response/p2p_advert_update_response_result.dart:997]: same here

File: lib/api/response/p2p_advert_list_response_result.dart [hunk 3]
```diff
@@ -401,88 +411,53 @@ abstract class ListItemModel {
...
+    super.remainingAmountDisplay,
+    super.visibilityStatus,
+  });
 
   /// Creates an instance from JSON.
   factory ListItem.fromJson(Map<String, dynamic> json) => ListItem(
         accountCurrency: json['account_currency'],
         advertiserDetails:
             AdvertiserDetails.fromJson(json['advertiser_details']),
+        blockTrade: getBool(json['block_trade'])!,
```
Comment (abedelaziz-deriv) [inline lib/api/response/p2p_advert_list_response_result.dart:460]: update unit tests for this logic

File: lib/api/response/p2p_advert_create_response_result.dart [hunk 4]
```diff
@@ -473,6 +449,7 @@ class P2pAdvertCreate extends P2pAdvertCreateModel {
             AdvertiserDetails.fromJson(json['advertiser_details']),
         amount: getDouble(json['amount'])!,
         amountDisplay: json['amount_display'],
+        blockTrade: getBool(json['block_trade'])!,
```
Comment (abedelaziz-deriv) [inline lib/api/response/p2p_advert_create_response_result.dart:452]: update unit tests for this logic

File: lib/api/response/p2p_advertiser_adverts_response_result.dart [hunk 5]
```diff
@@ -487,6 +459,7 @@ class ListItem extends ListItemModel {
             AdvertiserDetails.fromJson(json['advertiser_details']),
         amount: getDouble(json['amount'])!,
         amountDisplay: json['amount_display'],
+        blockTrade: getBool(json['block_trade'])!,
```
Comment (abedelaziz-deriv) [inline lib/api/response/p2p_advertiser_adverts_response_result.dart:462]: update unit tests for this logic

File: lib/api/response/p2p_advert_info_response_result.dart (hunk 1 above)
Comment (weng-deriv) [reply lib/api/response/p2p_advert_info_response_result.dart:1153]: @abedelaziz-deriv Upon checking, there's no places using this var. It's safe to keep it for now.

**Pull request merged**